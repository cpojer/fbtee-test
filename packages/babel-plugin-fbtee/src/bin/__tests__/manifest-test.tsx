import { join } from 'node:path';
import { generateManifest } from '../manifestUtils.tsx';

describe('manifest', () => {
  it('should extract strings', async () => {
    const srcPath = join(import.meta.dirname, '../__fixtures__');

    const { enumManifest, files } = await generateManifest(
      [srcPath],
      import.meta.dirname + '/../..',
    );

    expect(JSON.stringify(files, null, 2)).toMatchSnapshot();
    expect(JSON.stringify(enumManifest, null, 2)).toMatchSnapshot();
  });
});
