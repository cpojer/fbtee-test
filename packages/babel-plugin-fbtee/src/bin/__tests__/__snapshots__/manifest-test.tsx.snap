// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`manifest should extract strings 1`] = `
"[
  "bin/__fixtures__/TestFile.tsx"
]"
`;

exports[`manifest should extract strings 2`] = `
"{
  "ExportDefault$FbtEnum": {
    "id1": "groups",
    "id2": "photos",
    "id3": "videos"
  }
}"
`;
